### PR TITLE
fix bug of getBufferReverseIterator

### PR DESCRIPTION
**Pull request by futurist**
fix terminal getBufferReverseIterator issue, related to: https://github.com/xtermjs/xterm.js/issues/5378

### CODE REVIEW

**Review by DhruvTheDev1 — COMMENTED** [review]
Hi, nice fix to the getBufferReverseIterator()

So the root cause was a off by one error starting the loop `this.raw.buffer.active.length;` which is an invalid index instead of `this.raw.buffer.active.length - 1;` which prevents an out of bounds error and any incorrect behaviour.

File: src/vs/workbench/contrib/terminal/test/browser/xterm/xtermTerminal.test.ts [hunk 1]
```diff
@@ -79,7 +79,7 @@ const defaultTerminalConfig: Partial<ITerminalConfiguration> = {
 	fontWeight: 'normal',
 	fontWeightBold: 'normal',
 	gpuAcceleration: 'off',
-	scrollback: 1000,
+	scrollback: 10,
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/test/browser/xterm/xtermTerminal.test.ts]: Was this change needed?

**Review by Tyriar — CHANGES_REQUESTED** [review]

**Comment by Tyriar** [comment]
@futurist tests are failing

File: src/vs/workbench/contrib/terminal/test/browser/xterm/xtermTerminal.test.ts (hunk 1 above)
Comment (futurist) [reply src/vs/workbench/contrib/terminal/test/browser/xterm/xtermTerminal.test.ts]: yes it's needed, since need to test result of overflow of rows + scrollback limit.
